runtime/themes/gruvbox.toml

```diff
@@ -0,0 +1,83 @@
+# Author : Jakub Bartodziej <kubabartodziej@gmail.com>
+# The theme uses the gruvbox dark palette with standard contrast: github.com/morhetz/gruvbox
+
+"attribute" = "fg2"
+"keyword" = { fg = "red1" }
+"keyword.directive" = "red0"
+"namespace" = "yellow0"
+"punctuation" = "fg4"
+"punctuation.delimiter" = "fg4"
+"operator" = "orange0"
+"special" = "purple0"
+"property" = "fg2"
+"variable" = "fg2"
+"variable.builtin" = "fg3"
+"variable.parameter" = "fg2"
+"type" = "orange1"
+"type.builtin" = "orange0"
+"constructor" = "fg2"
+"function" = "green0"
+"function.macro" = "aqua1"
+"function.builtin" = "yellow1"
+"comment" = "gray1"
+"constant" = { fg = "fg2", modifiers = ["bold"] }
+"constant.builtin" = { fg = "fg1", modifiers = ["bold"] }
+"string" = "green1"
+"number" = "purple1"
+"escape" = { fg = "fg2", modifiers = ["bold"] }
+"label" = "aqua1"
+"module" = "yellow1"
+
+"warning" = "orange0"
+"error" = "red0"
+"info" = "purple0"
+"hint" = "blue0"
+
+"ui.background" = { bg = "bg0" }
+"ui.linenr" = { fg = "fg3" }
+"ui.linenr.selected" = { fg = "fg1", modifiers = ["bold"] }
+"ui.statusline" = { fg = "fg2", bg = "bg2" }
+"ui.statusline.inactive" = { fg = "fg3", bg = "bg4" }
+"ui.popup" = { bg = "bg1" }
+"ui.window" = { bg = "bg1" }
+"ui.help" = { bg = "bg1", fg = "fg1" }
+"ui.text" = { fg = "fg1" }
+"ui.text.focus" = { fg = "fg1", modifiers = ["bold"] }
+"ui.selection" = { bg = "bg3" }
+"ui.cursor.primary" = { bg = "bg3" }
+"ui.cursor.match" = { bg = "bg4" }
+"ui.menu" = { fg = "fg1" }
+"ui.menu.selected" = { fg = "fg3", bg = "bg3" }
+
+"diagnostic" = { modifiers = ["underlined"] }
+
+[palette]
+bg0 = "#282828" # main background
+bg1 = "#3c3836"
+bg2 = "#504945"
+bg3 = "#665c54"
+bg4 = "#7c6f64"
```

Comment:
Why 5 palettes color?

Reply:
The [gruvbox dark](https://github.com/morhetz/gruvbox#dark-mode-1) palette includes 10 neutral colors, 5 for use in foreground and 5 for use in background. The background colors are ordered from darkest to lightest, and the foreground colors are ordered from lightest to darkest. Using the i-th foreground color with the i-th background color results in a color composition whose contrast is balanced, and visually consistent with other i-th color pairs.

Those pairs give the designer 5 choices for making UI elements that distinguish themselves from their background and from each other. This property is useful when working with menus, pop up boxes, notification bars, status bars, etc. Each of these UI elements can be presented in one of 5 distinct visual characters while maintaining cohesion. This adds depth and diversity to the theme.

Reply:
It is probably best to stick to the color naming of gruvbox as you did here.

Reply:
I wonder why is it based on gruvbox when there are many other popular color platelets like base16. Also, not sure why gray only have 2, what about blue gray?

Reply:
It's based on gruvbox because I like it. It's a slightly more popular palette than base16 (9.3k stars vs 8k), so obviously many people like it as well. The whole point of themes is everyone use their preferred palette to make a theme. I didn't change the palette because then it woudn't be gruvbox anymore.